x/epoching/abci.go

```diff
@@ -89,6 +90,20 @@ func EndBlocker(ctx context.Context, k keeper.Keeper) ([]abci.ValidatorUpdate, e
 		queuedMsgs := k.GetCurrentEpochMsgs(ctx)
 		// forward each msg in the msg queue to the right keeper
 		for _, msg := range queuedMsgs {
+			msgId := hex.EncodeToString(msg.MsgId)
+
+			unlockCtx, unlockCommit := sdkCtx.CacheContext()
```

Comment:
Please logically compare the implications of having unlockCtx versus not having it, and re-evaluate whether it is truly necessary